### ISSUE TITLE
Doc/Feature request: Use with flatbuffers?

### ISSUE THREAD

**Issue by shaunc** [improbable-eng/grpc-web#376]
I have go code using flatbuffers and am designing a browser-based visualization tool. Are there examples of using grpc-web using flatbuffers anywhere? Thanks!

**Comment by johanbrandhorst** [comment]
Nope! I think that's very rare on the ground unfortunately. This library doesn't really allow you to configure the wire format either, so you'll have to find a JS flatbuffer marshaller and inject it into the client somehow. You probably want to use a custom content type header in accordance with the spec (may I suggest `application/grpc-web+flatbuffers`? It might be possible to hardcode a parser into the existing client (like we have one for base64 `application/grpc-web-text`, but the best solution would be to add something like a `Codec` interface that could be implemented by any wire encoding. See the go gRPC library `Codec` for inspiration.

Would you be interested in contributing an implementation of this?

**Comment by shaunc** [comment]
Thanks for the reply! ... hmm ... well I'm "interested" but we're just in the design phase now, and there are other things we will prototype/experiment first before circling back to this. Probably we'll use some other way to connect to start out, then circle back to reconsider this (ETA -- summer).
